packages/teko/src/Tpetra/Teko_BlockingTpetra.cpp

```diff
@@ -250,6 +249,34 @@ RCP<Tpetra::CrsMatrix<ST,LO,GO,NT> > buildSubBlock(int i,int j,const RCP<const T
    std::vector<GO> colIndices(maxNumEntries);
    std::vector<ST> colValues(maxNumEntries);
 
+   // for counting
+   std::vector<size_t> nEntriesPerRow(numMyRows);
+
+   // insert each row into subblock
+   // Count the number of entries per row in the new matrix
+   for(LO localRow=0;localRow<numMyRows;localRow++) {
+      size_t numEntries = -1;
+      GO globalRow = gRowMap->getGlobalElement(localRow);
+      LO lid = A->getRowMap()->getLocalElement(globalRow);
+      TEUCHOS_ASSERT(lid>-1);
```

Comment:
In the future, it's nicer to use the flag value `Teuchos::OrdinalTraits<LO>::invalid()`.